homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,90 @@
 from typing import Any
 
 from aiohttp import ClientConnectionError
-from intellifire4py import IntellifireAsync
+from intellifire4py import AsyncUDPFireplaceFinder, IntellifireAsync
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
-from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DOMAIN
+from .const import DOMAIN, LOGGER
 
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 
 
-async def validate_input(hass: HomeAssistant, host: str) -> str:
+async def validate_host_input(host: str) -> str:
     """Validate the user input allows us to connect.
 
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     api = IntellifireAsync(host)
     await api.poll()
-
+    ret = api.data.serial
+    LOGGER.info("Found a fireplace: %s", ret)
     # Return the serial number which will be used to calculate a unique ID for the device/sensors
-    return api.data.serial
+    return ret
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for IntelliFire."""
 
-    VERSION = 1
+    VERSION = 2
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
-            )
+    def __init__(self):
+        """Initialize the Config Flow Handler."""
+        self._discovered_host: str = ""
+        self.fireplace_finder = AsyncUDPFireplaceFinder()
+        self._config_context = {}
```

Comment:
This specific PR could probably get away without using `_config_context` when I get into a multi-step flow it will be VERY important. I'm hoping you'll allow it to remain in this PR ... but if not I can pull it out and then put it back in later...

Reply:
I was told that `__init__` is not always called -> and @bdraco suggested I move this to another function call

Reply:
For context `AsyncUDPFireplaceFinder` isn't always used so we don't want to create it when it is not going to be used.  `__init__` is always called when creating the object.

Reply:
Makes total sense! :) (updating it now (just need to verify the tests still work!)